src/preprocessing.py

```diff
@@ -33,18 +34,24 @@ def preprocessing(df):
         'security_test_result (FALSE=Passed; TRUE=Failed)': 'security_test_result',
         'risk_label_Baljeet': 'risk_label'
         }, inplace=True)
+    # Merge country_metric data with input data
+    df = df.merge(country_metric_df, left_on='server_location', right_on='Country')
```

Comment:
I used left join to preserve rows with missing country field
```suggestion
    df = df.merge(country_metric_df, left_on='server_location', right_on='Country', how='left')
```